src/MultiFormats/NetworkProtocol.cs

```diff
@@ -0,0 +1,580 @@
+    {
+        public ushort Port { get; set; }
+        public override string Name => "tcp";
+        public override uint Code => 6;
+
+        public override void ReadValue(TextReader stream)
+        {
+            base.ReadValue(stream);
+            try
+            {
+                Port = ushort.Parse(Value);
+            }
+            catch (Exception e)
+            {
+                throw new FormatException(string.Format("'{0}' is not a valid port number.", Value), e);
+            }
+        }
+
+        public override void ReadValue(CodedInputStream stream)
+        {
+            var bytes = stream.ReadSomeBytes(2);
+            Port = (ushort) IPAddress.NetworkToHostOrder(BitConverter.ToInt16(bytes, 0));
+            Value = Port.ToString(CultureInfo.InvariantCulture);
+        }
+
+        public override void WriteValue(CodedOutputStream stream)
+        {
+            var bytes = BitConverter.GetBytes(IPAddress.HostToNetworkOrder((short) Port));
+            stream.WriteSomeBytes(bytes);
+        }
+    }
+
+    internal class UdpNetworkProtocol : TcpNetworkProtocol
+    {
+        public override string Name => "udp";
+        public override uint Code => 273;
+    }
+
+    internal class DccpNetworkProtocol : TcpNetworkProtocol
+    {
+        public override string Name => "dccp";
+        public override uint Code => 33;
+    }
+
+    internal class SctpNetworkProtocol : TcpNetworkProtocol
+    {
+        public override string Name => "sctp";
+        public override uint Code => 132;
+    }
+
+    internal abstract class IpNetworkProtocol : NetworkProtocol
+    {
+        public IPAddress Address { get; set; }
+
+        public override void ReadValue(TextReader stream)
+        {
+            base.ReadValue(stream);
+            try
+            {
+                // Remove the scope id.
+                var i = Value.LastIndexOf('%');
+                if (i != -1)
+                    Value = Value.Substring(0, i);
+
+                Address = IPAddress.Parse(Value);
+            }
+            catch (Exception e)
+            {
+                throw new FormatException(string.Format("'{0}' is not a valid IP address.", Value), e);
+            }
+        }
+
+        public override void WriteValue(TextWriter stream)
+        {
+            stream.Write('/');
+            stream.Write(Address.ToString());
+        }
+
+        public override void WriteValue(CodedOutputStream stream)
+        {
+            var ip = Address.GetAddressBytes();
+            stream.WriteSomeBytes(ip);
+        }
+    }
+
+    internal class Ipv4NetworkProtocol : IpNetworkProtocol
+    {
+        private static int _addressSize = IPAddress.Any.GetAddressBytes().Length;
+
+        public override string Name => "ip4";
+        public override uint Code => 4;
+
+        public override void ReadValue(TextReader stream)
+        {
+            base.ReadValue(stream);
+            if (Address.AddressFamily != AddressFamily.InterNetwork)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)